packages/main/src/Dialog.hbs

```diff
@@ -8,7 +8,11 @@
 			tabindex="{{_headerTabIndex}}"
 			@keydown="{{_onDragOrResizeKeyDown}}"
 			@mousedown="{{_onDragMouseDown}}"
-			part="header">
+			part="header"
+			state="{{this.state}}">
+			{{#if hasValueState}}
+			<ui5-icon class="ui5-dialog-value-state-icon" name="{{_dialogStateIcon}}"></ui5-icon>
```

Comment:
This may result in broken UI when people use custom headers. Can you investigate how people commonly supply custom headers when using the header slot and see if this will break it? Maybe we should have another approach to displaying an icon when the slot="header" is supplied.

Reply:
In the case, when slot="header" is used, indeed the state won't be reflected with an icon as it's done with the built-in header. I can see 2 options: 

-  **Less work for the apps:** to introduce a new DOM element, that will wrap the relevant icon + the custom header slot - to place the icon according to the spec, give it left and right offset according to the spec and leave the rest of the horizontal space for the custom header:
```html
{{#if header.length }}
   <div>
    <ui5-icon class="ui5-dialog-value-state-icon" name="{{_dialogStateIcon}}"></ui5-icon>
     <slot name="header"></slot>
   </div>
```

-  **More work for the apps:**. To document this and suggest the app developers render an icon, place it in front of their title and switch it if needed

I think we should try with the first option.

Reply:
What if there is a Bar as a slotted element for the header? It relies on its own contentStart, middle, contentEnd slots to place its children elements.

Reply:
The left part will be reserved for the state icon and the rest of the horizontal space for the Bar's contentStart, middle, contentEnd slots. Perhaps we should test a bit and see how it works.

Reply:
If we consider that the Belize theme will soon be deprecated, the Title component then will always be placed in contentStart - left aligned (per Fiori Guidelines). That would look okay, except for the margin-inline-start of the icon, which won't be correct according to the design spec. The spec defines margin-right: 0.5rem from a title of a dialog.

Reply:
I would not consider Belize at all. The second one sounds problematic, I would also set a part for the state icon, so that people can clear the offset if they are using a Bat, that comes with it's own padding.

Reply:
We have come to an agreement to leave it as is for now and print the state icon regardless of the header type. Also setting a part for the state icon will require more complex calculations from the app developers.